enhancements/etcd/automated-backups.md

```diff
@@ -0,0 +1,376 @@
+        // Wait for the current one to complete?
+    Schedule string `json:"schedule"`
+    
+    // RetentionPolicy defines the retention policy for retaining and deleting existing backups.
+    // +optional
+    RetentionPolicy RetentionPolicy `json: "retentionPolicy"`
+}
+
+// RetentionPolicy defines the retention policy for retaining and deleting existing backups.
+// This struct is a discriminated union that allows users to select the type of retention policy from the supported types.
+// +union
+type RetentionPolicy struct {
+  // RetentionType sets the type of retention policy. The currently supported and valid values are "retentionCount"
+	// Currently, the only valid policies are retention by count (RetentionCount) and by size (RetentionSizeGb). More policies or types may be added in the future.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+  // +kubebuilder:validation:Enum:=RetentionCount;RetentionSizeGb
+	RetentionType RetentionType `json:"retentionType"`
+	
+  // RetentionCount defines the maximum number of backups to retain.
+  // If the number of successful backups matches retentionCount 
+  // the oldest backup will be removed before a new backup is initiated.
+  // The count here is for the total number of backups
+  // +kubebuilder:validation:Minimum=1
+  // +optional
+  RetentionCount int `json: "retentionCount,omitempty"`
+
+  // RetentionSizeGb defines the total size in Gb of backups to retain.
+  // If the current total size backups exceeds RetentionSizeGb then 
+  // the oldest backup will be removed before a new backup is initiated.
+  // +kubebuilder:validation:Minimum=0.1
+  // +optional
+  RetentionSizeGb float64 `json: "retentionSizeGb,omitempty"`
+
+  // TODO: Can the union members be unspecified, and if so what would they default to?
+  // Add // +kubebuilder:default=<value>
+	
+}
+
+// RetentionType is the enumeration of valid retention policy types
+// +enum
+// +kubebuilder:validation:Enum:="RetentionCount";"RetentionSizeGb"
+type RetentionType string
+
+const (
+  RetentionTypeCount RetentionType = "RetentionCount"
+  RetentionTypeSize RetentionType = "RetentionSizeGb"
+)
+
+```
+
+#### EtcdBackupRequest API
+
+The `EtcdBackupRequest` CRD will be introduced to the API group `backup.etcd.openshift.io` with the version `v1alpha1`. The CRD would be feature-gated with the annotation `release.openshift.io/feature-set: TechPreviewNoUpgrade` until the prerequisite e2e test has an acceptable pass rate.
+
+The spec and status will be as listed below:
+
+```Go
+// EtcdBackupRequestSpec represents the configuration of the EtcdBackupRequest CRD.
+type EtcdBackupRequestSpec struct {
+
+    // Reason defines the reason for the most recent backup.
+    // TODO: Do we need to specify a reason? This is more metadata than spec to reconcile.
+    // We could just have a spec-less CR as created by the admin or the EtcdBackupSchedule per the schedule.
+    // The reason could be tagged into an annotation if needed.
```

Comment:
we don't use annotations for structured fields.  Reason is entirely reasonable to include as optional here.